streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -109,6 +109,18 @@ abstract class DStream[T: ClassTag] (
   /* Set the creation call site */
   private[streaming] val creationSite = DStream.getCreationSite()
 
+  /**
+   * The scope associated with the operation that created this DStream.
+   *
+   * This is the medium through which we pass the DStream operation name (e.g. updatedStateByKey)
+   * to the RDDs created by this DStream. Note that we never directly use this scope in RDDs.
+   * Instead, every time we call `compute` we instantiate a new scope using the same name as this
+   * one. Otherwise, all RDDs ever created by this DStream will be in the same scope.
+   */
+  private val scope: Option[RDDOperationScope] = {
```

Comment:
Can you store only the name in as a DStream field? I does not make sense to save this arbitrary mostly unnecessary RDDOperationScope object to be serialized with DStreams as part of DStream checkpoints.